docs/README.md

```diff
@@ -0,0 +1,78 @@
+# Incremental ASP.NET to ASP.NET Core Migration
+
+Migrating an application from ASP.NET Framework to ASP.NET Core is a non-trivial action for the majority of large applications. These applications have often grown organically over time, incorporating new technologies as they come and are often composed of many legacy decisions. The aim of this repo is to provide tools and support for migrating these large applications with as little change as possible to ASP.NET Core.
+
+One of the larger challenges is the pervasive use of `System.Web.HttpContext` throughout a code base to access information about a request or update a response. Until now, in order to move forward, a large scale rewrite was required to remove this dependency. Now, the adapters in this repo provide a set of runtime helpers to access the types your code is expecting but in a way that will work on ASP.NET Core with little change.
+
+A complete migration may take a while (sometimes a multi-year effort) depending on the size of the application. In order to continue deploying an application to production while working on migrating, the best pattern is to follow is the [Strangler Fig pattern](https://docs.microsoft.com/en-us/azure/architecture/patterns/strangler-fig). This pattern allows for continual development on the old system with an incremental approach to moving forward. This document will describe how to apply that pattern to an ASP.NET app migrating towards ASP.NET Core.
+
+To jump into the process, please see the [Getting Started](getting_started.md) guide.
+
+## Migration Journey
+
+When starting a migration journey, the application will be targeting ASP.NET Framework and running on Windows with its supporting libraries:
+
+```mermaid
+flowchart LR;
+  external[External Traffic] --> framework[.NET Framework Application]
+  framework --- libraries[[Business logic]]
+```
+
+The first step is to introduce a new application based on ASP.NET Core that will become the entry point. Traffic will enter the core app and if the core app cannot match a route, it will proxy the request onto the .NET Framework Application via [YARP](https://microsoft.github.io/reverse-proxy/) and serve a response as the application has already been doing. Majority of code will continue to be in the .NET Framework Application, but the core app is now set up to start migrating routes to:
+
+```mermaid
+flowchart LR;
+  external[External Traffic] --> core[ASP.NET Core Application]
+  core --- libraries
+  core -- YARP proxy --> framework[ \n\n.NET Framework Application\n\n\n]
+  framework --- libraries[[Business logic]]
+```
+
+In order to start moving over business logic that relies on `HttpContext`, the libraries need to be built against `Microsoft.AspNetCore.SystemWebAdapters`. This will ensure that the libraries are using surface area that is available on both ASP.NET and ASP.NET Core:
```

Comment:
I like to include the fact that using SystemWebAdapters can enable libraries to target .NET Standard since that's a known way of making libraries sharable between NetFx and NetCore callers.